plugins/newsdownloader.koplugin/main.lua

```diff
@@ -82,6 +83,22 @@ function NewsDownloader:addToMainMenu(menu_items)
                 callback = function() self:removeNewsButKeepFeedConfig() end,
             },
             {
+                text = _("Turn on/off WiFi"),
```

Comment:
I'm not sure what the purpose of this submenu is?

Reply:
It's shortcut. In android top menu is not visible so it's faster to turn it here than go to settings -> fullscreen -> disable -> enable disable Wifi in android top menu -> revert the settings to fullscreen.

Reply:
There's no need for such duplication.

Click `Download news` (or whatever) → wifi prompt

To turn off `Settings` → Network → turn off

The Android top menu doesn't come into play.

Reply:
Anyway I guess >90% use cases for News downloader you would like to turn on wifi before operation and turn it off after. 

There could be a prompt before downloading (contained in this code change) [ 100% users need that - as it won't work without internet] 
and after (I thought some users might be annoyed by that, that's why decided to do a separte menu)

Reply:
@Frenzie it's much faster to do it from *current* submenu than going to "Settings"

Reply:
ok. replaced with prompt before and after downloading